synfig-studio/src/gui/dialogs/dialog_pluginmanager.cpp

```diff
@@ -179,6 +186,58 @@ Dialog_PluginManager::build_listbox()
     plugin_list_box.show_all_children();
 }
 
+std::string extract_plugin_name(std::istream &fileStream)
+{
+    xmlpp::DomParser parser;
+    parser.set_substitute_entities();
+    parser.parse_stream(fileStream);
+
+    const xmlpp::Node* pNode = parser.get_document()->get_root_node();
+    if (std::string(pNode->get_name()) != std::string("plugin") ) {
+        synfig::warning("Invalid plugin.xml file (missing root <plugin>)");
+        return std::string("");
+    }
+
+    auto execlist = pNode->find("./exec");
+    if (!execlist.empty())
+    {
+        for (const xmlpp::Node* node : pNode->find("./name") ) {
+            const xmlpp::Element* element = dynamic_cast<const xmlpp::Element*>(node);
+            std::string lang = element->get_attribute_value("lang");
+            if (lang.empty()) {
+                return element->get_child_text()->get_content();
+            }
+        }
+    }
+}
+std::string get_plugin_name(etl::handle<synfig::FileContainerZip> &zip_fs, const PluginZipStatus &pluginStatus, const std::vector<std::string> &files) {
+    switch (pluginStatus) {
+        case PLUGIN_ZIP_AT_ROOT_DIR:
+            for (const auto& file : files) {
+                if (file == "plugin.xml") {
+                        return extract_plugin_name(*zip_fs->get_read_stream(file));
```

Comment:
Every time you deference a (smart or not) pointer, check if it isn't null before you do it :)
Otherwise, ... crash is possible ;)